### ISSUE TITLE
bitstamp trader fails to buy

### ISSUE THREAD

**Issue by jonny64** [askmike/gekko#266]
apparently bitstamp trader is somehow broken
```
C2015-02-03 02:41:40 (INFO):	Attempting too BUY NaN BTC at Bitstamp
j2015-02-03 02:41:41 (ERROR):	unable to buy: null { error: 'Missing amount and/or price POST parameters' }
```

gekko build from 48fe8b (current master)
`C2015-02-03 02:41:40 (INFO):	Attempting too BUY `
misprint `too BUY`

**Comment by askmike** [comment]
this is fixed in 0.2.

**Comment by ghost** [comment]
Today when the Gekko Bot tried to register a transaction in BitStamp gave exactly the same mistake

```
 **(ERROR): Unable to buy: null {error: 'Missing amount and / or price POST parameters'}**
```

Before starting the Gekko run commands

```
 **git pull && npm update**
```

Is there anything you can do to fix the problem?

**Comment by askmike** [comment]
Could you check which version of the `bitstamp` module is installed?

```
npm list | grep bitstamp
```

Also, did you have debug enabled and log outputs just before this one (with the amount to buy)?

**Comment by ghost** [comment]
First of all...THANKS to answer so quickly! :)

The information that is provided by the command indicated is as follows:

bitstamp@0.1.10

**Comment by askmike** [comment]
That is an outdated version. Did run `npm update`? Strange..

Anyway, to fix, run:

   npm remove bitstamp
   npm install bitstamp

**Comment by ghost** [comment]
I tried what you suggest Mike but without sucess...

When i do

npm remove bitstamp
npm install bitstamp

everytime i do this...it install version 0.1.10 However do get a more recent version i use the command   **npm install bitstamp@latest**  and with this i get the version 0.3.0 (as you can see)

```
pi@raspberrypi:~ $ sudo npm list | grep bitstamp
└─┬ bitstamp@0.3.0

HOWEVER (sorry...) now when i start gekko i get this error...

pi@raspberrypi:~ $ cd gekko/
pi@raspberrypi:~/gekko $ sudo node gekko
2016-06-16 19:58:47 (INFO):     Gekko v0.1.2 started
2016-06-16 19:58:47 (INFO):     I'm gonna make you rich, Bud Fox.


2016-06-16 19:58:47 (INFO):     Using config file: /home/pi/gekko/config.js
2016-06-16 19:58:47 (INFO):     Setting up Gekko in realtime mode

2016-06-16 19:58:48 (INFO):     Setting up:
2016-06-16 19:58:48 (INFO):              Trading Advisor
2016-06-16 19:58:48 (INFO):              Calculate trading advice
2016-06-16 19:58:48 (INFO):              Using trading method: DEMA

2016-06-16 19:58:48 (INFO):     Setting up:
2016-06-16 19:58:48 (INFO):              Trader
2016-06-16 19:58:48 (INFO):              Trader will follow the advice and create real orders.
_http_client.js:51
    throw new TypeError('Request path contains unescaped characters.');
    ^

TypeError: Request path contains unescaped characters.
    at new ClientRequest (_http_client.js:51:11)
    at Object.exports.request (http.js:31:10)
    at Object.exports.request (https.js:163:15)
    at Bitstamp._request (/home/pi/gekko/node_modules/bitstamp/bitstamp.js:40:19)
    at Bitstamp._post (/home/pi/gekko/node_modules/bitstamp/bitstamp.js:139:8)
    at Bitstamp.balance (/home/pi/gekko/node_modules/bitstamp/bitstamp.js:190:8)
    at Trader.getPortfolio (/home/pi/gekko/exchanges/bitstamp.js:55:17)
    at bound [as getPortfolio] (/home/pi/gekko/node_modules/lodash/dist/lodash.js:729:21)
    at Manager.setPortfolio (/home/pi/gekko/core/portfolioManager.js:80:17)
    at bound (/home/pi/gekko/node_modules/lodash/dist/lodash.js:729:21)
```

what is wrong now ?

(I need to pay at least 1 beer to you by your help :+1:

**Comment by askmike** [comment]
I should have been more clear, but you have to be sure to run `npm install bitstamp` (or `npm install bitstamp@latest`) inside the `gekko` directory. Could you make sure it is installed there as well?

**Comment by ghost** [comment]
I run all comand inside GEKKO directory

```
`pi@raspberrypi:~/gekko $ ls
bin        docs          gekko.js  methods       plugins     scripts
config.js  exchanges     history   node_modules  plugins.js  subscriptions.js
core       exchanges.js  log       package.json  README.md   test
pi@raspberrypi:~/gekko $ sudo npm install bitstamp@latest
npm WARN package.json gekko@0.1.2 No license field.
bitstamp@0.3.0 node_modules/bitstamp
└── underscore@1.4.4
pi@raspberrypi:~/gekko $ node gekko
2016-06-16 20:49:07 (INFO):     Gekko v0.1.2 started
2016-06-16 20:49:07 (INFO):     I'm gonna make you rich, Bud Fox.


2016-06-16 20:49:07 (INFO):     Using config file: /home/pi/gekko/config.js
2016-06-16 20:49:07 (INFO):     Setting up Gekko in realtime mode

2016-06-16 20:49:07 (INFO):     Setting up:
2016-06-16 20:49:07 (INFO):              Trading Advisor
2016-06-16 20:49:07 (INFO):              Calculate trading advice
2016-06-16 20:49:08 (INFO):              Using trading method: DEMA

2016-06-16 20:49:08 (INFO):     Setting up:
2016-06-16 20:49:08 (INFO):              Trader
2016-06-16 20:49:08 (INFO):              Trader will follow the advice and create real orders.
_http_client.js:51
    throw new TypeError('Request path contains unescaped characters.');
    ^

TypeError: Request path contains unescaped characters.
    at new ClientRequest (_http_client.js:51:11)
    at Object.exports.request (http.js:31:10)
    at Object.exports.request (https.js:163:15)
    at Bitstamp._request (/home/pi/gekko/node_modules/bitstamp/bitstamp.js:40:19)
    at Bitstamp._post (/home/pi/gekko/node_modules/bitstamp/bitstamp.js:139:8)
    at Bitstamp.balance (/home/pi/gekko/node_modules/bitstamp/bitstamp.js:190:8)
    at Trader.getPortfolio (/home/pi/gekko/exchanges/bitstamp.js:55:17)
    at bound [as getPortfolio] (/home/pi/gekko/node_modules/lodash/dist/lodash.js:729:21)
    at Manager.setPortfolio (/home/pi/gekko/core/portfolioManager.js:80:17)
    at bound (/home/pi/gekko/node_modules/lodash/dist/lodash.js:729:21)
```

`

**Comment by askmike** [comment]
Very strange. This line:

```
2016-06-16 20:49:07 (INFO):     Using config file: /home/pi/gekko/config.js
```

Does not come from my version of Gekko (did you add that?). Could you post the output of the following commands?

```
node gekko --version
npm --version
git log -10
```

**Comment by ghost** [comment]
`pi@raspberrypi:~/gekko $ node gekko.js --version
Gekko version: v0.1.2
Nodejs version: v4.2.1
node.js:756
        throw new TypeError('invalid pid');
        ^

TypeError: invalid pid
    at process.kill (node.js:756:15)
    at Object.util.die (/home/pi/gekko/core/util.js:141:13)
    at Object.<anonymous> (/home/pi/gekko/gekko.js:31:8)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:134:18)
    at node.js:961:3
pi@raspberrypi:~/gekko $ npm --version
3.9.5
pi@raspberrypi:~/gekko $ git log -1
commit e7ab4254fc3940d32822e62690dbfc5d9cb2019c
Merge: 0ebbb24 57fa062
Author: Mike van Rossum mike@mikevanrossum.nl
Date:   Sun Jun 5 12:20:08 2016 +0200

```
Merge pull request #281 from meXBT/williamcoates-readme-update

Added meXBT to list of exchanges
```

`

**Comment by askmike** [comment]
ahhh, your gekko is not up to date (for some reason), sorry for all the confusion. To fix, make sure you git pull inside the gekko directory (and that you are on the `stable` branch):

```
git fetch origin
git checkout stable
git pull
```

(or just redownload from github)

Here is a timeline of events:
1. Bitstamp introduces v2 API.
2. Gekko changes branch model and the old main branch was deleted (this means that all "git pulls" to update do not work anymore)
3. Bitstamp npm module get's updated to support the v2 API (version 0.3, at this time Gekko still installs older version of the module).
4. Gekko gets updated and now includes bitstamp@0.3.x
5. This PR gets closed because the original issue is not present in the v2 API.

But after you commented here I was assuming you were on the latest version of gekko.

**Comment by ghost** [comment]
Probably i do something wrong now i cant even start GEKKO :P

Today at night i will install again GEKKO in a fresh install of OS :)

I will give news and again...a lot of thanks for your patience

**Comment by askmike** [comment]
sounds great, if you have any other problem please create a new issue.
